lib/mfs.js

```diff
@@ -0,0 +1,125 @@
+var stream = require('stream')
+
+module.exports = function (node) {
+  var store = {}
+  var baseDir = '/npm-registry/'
+  store.baseDir = baseDir
+  store.node = node
+
+  node.files.mkdir(baseDir, { p: true }, function (err) {
+    if (err) {
+      console.error(err)
+    }
+  })
+
+  store.createWriteStream = function (opts, cb) {
+    if (typeof opts === 'string') opts = {key: opts}
+    if (opts.name) opts.key = opts.name
+    if (!cb) cb = noop
+
+    var bufferStream = new stream.PassThrough()
+    var buffer
+
+    bufferStream.on('data', function (chunk) {
+      if (!buffer) buffer = chunk
+      else {
+        buffer = Buffer.concat([buffer, chunk], buffer.length + chunk.length)
+      }
+    })
+
+    bufferStream.on('end', function () {
+      // nested, must make sure we have all the dirs along the way
+      if (opts.key.indexOf('/')) {
```

Comment:
Should be checking for -1 right? This will fail if the key is an absolute path, and will successes even if no forward slash is given, resulting in a directory named after your key.